src/apps/authentication/serializers.py

```diff
@@ -55,3 +56,33 @@ def set_user_role(self, user, role):
         if role and role != UserRole.ADMIN:
             user.role = role
             user.save()
+
+
+class UserLoginSerializer(serializers.Serializer):
+    id = serializers.UUIDField(read_only=True)
+    email = serializers.EmailField()
+    username = serializers.CharField(read_only=True)
+    password = serializers.CharField(write_only=True)
+    access_token = serializers.CharField(read_only=True)
+
+    def validate(self, attrs):
+        email = attrs.get("email")
+        password = attrs.get("password")
+
+        if email and password:
+            user = User.objects.get(email=email)
+            if not user:
+                raise serializers.ValidationError("User not found")
+
+            if not user.check_password(password):
+                raise serializers.ValidationError("Incorrect password")
+
+            refresh_token = RefreshToken.for_user(user)
+
+            attrs["access_token"] = str(refresh_token.access_token)
+            attrs["id"] = user.id
+            attrs["username"] = user.username
```

Comment:
```suggestion
            attrs["user"] = user
```

Wouldn't it be better to transfer the entire user?